File: c#-solution/0168. Excel Sheet Column Title.cs
// 168. Excel Sheet Column Title
// https://leetcode.com/problems/excel-sheet-column-title/
// Tag: C#, HashTable, String, Easy

// 1. Brute Force
public class Solution {
    public string ConvertToTitle(int columnNumber) {
        var dic = Enumerable.Range(0, 26).ToDictionary(i => i+1, i=> (char)('A' + i));

        //Console.WriteLine(String.Join(",", dic.Select(c => $"{c.Key}:{c.Value}")));
        var arr = new List<char>();
        int col = columnNumber;
        while(col > 0){
            int temp  = col % 26;
            if(temp == 0) temp =26;
            arr.Add(dic[temp]);
            col = (col - temp) /26;
        }
        arr.Reverse();
        return new String(arr.ToArray());
    }
}
// TC: O(log n)
// SC: O(log n)

File: problem/0168. Excel Sheet Column Title.js
// 168. Excel Sheet Column Title
// https://leetcode.com/problems/excel-sheet-column-title/
// Tag: Math, String, Easy

/**
 * @param {number} columnNumber
 * @return {string}
 */

// brute force
var convertToTitle = function(columnNumber) {
    let str = '';
    let n = columnNumber;
    while(n > 0){
        let temp = n % 26;
        if(temp === 0){
            str = 'Z' + str;
            n = Math.floor(n/26) -1;
        } else {
            let char = String.fromCharCode(64 + temp);
            str = char + str;
            n = Math.floor(n/26);
        }
    }
    return str;
};
// TC : O(log26(n))
// SC : O(1)
